### PR TITLE
Theme documentation to Paragraph component

### PR DESCRIPTION
**Pull request by ShimiSun**
#### What does this PR do?
Adding theme documentation to Paragraph component

#### Where should the reviewer start?
src/js/components/Paragraph/doc.js

#### What testing has been done on this PR?
storybook

#### How should this be manually tested?
changing the theme to apply color and text size to paragraph

#### Any background context you want to provide?
No

#### What are the relevant issues?
None

#### Screenshots (if appropriate)
N/A

#### Do the grommet docs need to be updated?
No

#### Should this PR be mentioned in the release notes?
No

#### Is this change backward compatible or is it a breaking change?
backward compatible

### CODE REVIEW

File: src/js/components/Paragraph/doc.js [hunk 1]
```diff
@@ -29,3 +29,21 @@ export const doc = Paragraph => {
...
+    description: 'The text color used inside the Paragraph.',
+    type: 'string | { dark: string, light: string }',
+    defaultValue: "{ dark: '#f8f8f8', light: '#444444' }",
+  },
+  'global.text': {
+    description: 'The text size of the Paragraph text.',
+    type: 'string',
+    defaultValue: 'medium',
+  },
+  'paragraph.extend': {
```
Comment (ericsoderberghp) [inline src/js/components/Paragraph/doc.js]: We also need to include `paragraph.<size>` from base.js, which controls the font-size, line-height, and max-width of the paragraph.

File: src/js/components/Paragraph/doc.js [hunk 2]
```diff
@@ -29,3 +29,21 @@ export const doc = Paragraph => {
 
   return DocumentedParagraph;
 };
+
+export const themeDoc = {
+  'global.colors.text': {
+    description: 'The text color used inside the Paragraph.',
+    type: 'string | { dark: string, light: string }',
+    defaultValue: "{ dark: '#f8f8f8', light: '#444444' }",
+  },
```
Comment (ericsoderberghp) [inline src/js/components/Paragraph/doc.js]: We also need to include `global.edgeSize.<size>` and `global.edgeSize.responsiveBreakpoint`, which are used in `genericStyles()` within StyledParagraph.

Comment (ShimiSun) [reply src/js/components/Paragraph/doc.js]: Done. 
Not 100% sure about the global.edgeSize.<size>, and on whether it should document all sizes. Let me know what you think.

File: src/js/components/Paragraph/doc.js (hunk 1 above)
Comment (ShimiSun) [reply src/js/components/Paragraph/doc.js]: Done. Let me know if it is compliant with what you had in mind.

File: src/js/components/Paragraph/doc.js [hunk 3]
```diff
@@ -29,3 +29,63 @@ export const doc = Paragraph => {
...
+  'global.text': {
+    description: 'The text size of the Paragraph text.',
+    type: 'string',
+    defaultValue: 'medium',
+  },
+  'global.edgeSize.small': {
```
Comment (ericsoderberghp) [inline src/js/components/Paragraph/doc.js]: I don't think we need this as well as `global.edgeSize` below. I recommend dropping this one and keeping `global.edgeSize`.

File: src/js/components/Paragraph/doc.js [hunk 4]
```diff
@@ -29,3 +29,63 @@ export const doc = Paragraph => {
...
+    description: 'The padding around paragraph.',
+    type: 'string',
+    defaultValue: '12px',
+  },
+  'global.edgeSize': {
+    description: 'The possible sizes for gap, margin, and pad.',
```
Comment (ericsoderberghp) [inline src/js/components/Paragraph/doc.js]: There is no `gap` or `padding` for a Paragraph. So, this is just for `margin`.

File: src/js/components/Paragraph/doc.js [hunk 5]
```diff
@@ -29,3 +29,63 @@ export const doc = Paragraph => {
...
+    xsmall: '6px',
+    small: '12px',
+    medium: '24px',
+    large: '48px',
+    xlarge: '96px',
+    responsiveBreakpoint: 'small',
+  },
+}`,
+  },
+  'paragraph.maxWidth': {
```
Comment (ericsoderberghp) [inline src/js/components/Paragraph/doc.js]: I don't think we have a `paragraph.maxWidth`. We do have `paragraph.<size>.maxWidth`. I could see either adding that via `paragraph`, with an object showing each size, sort of like we have for `global.edgeSize` above. Or, we could add one per sub key, like `paragraph.<size>.maxWidth` and then show the maxWidth value for each size. I think the former would be a little more consistent, albeit a bit verbose.

Comment (ShimiSun) [reply src/js/components/Paragraph/doc.js]: I adopted the former way, it is more consistent and readable imo

File: src/js/components/Paragraph/doc.js [hunk 6]
```diff
@@ -29,3 +29,76 @@ export const doc = Paragraph => {
...
+  'global.text': {
+    description: 'The text size of the Paragraph text.',
+    type: 'string',
+    defaultValue: 'medium',
+  },
+  'global.edgeSize': {
+    description: 'The possible sizes for margin.',
+    type: 'object',
+    defaultValue: `{
+      edgeSize: {
```
Comment (ericsoderberghp) [inline src/js/components/Paragraph/doc.js]: I don't think we need to repeat `edgeSize` here, as that's already in the key name.

File: src/js/components/Paragraph/doc.js [hunk 7]
```diff
@@ -29,3 +29,76 @@ export const doc = Paragraph => {
...
+        xlarge: '96px',
+        responsiveBreakpoint: 'small',
+      },
+    }`,
+  },
+  paragraph: {
+    description: `The maximum width, font-size and line height of the paragraph.`,
+    type: 'string',
+    defaultValue: `{
+      size: {
```
Comment (ericsoderberghp) [inline src/js/components/Paragraph/doc.js]: There is no `size` key anywhere under the paragraph key.

File: src/js/components/Paragraph/doc.js [hunk 8]
```diff
@@ -29,3 +29,76 @@ export const doc = Paragraph => {
...
   return DocumentedParagraph;
 };
+
+export const themeDoc = {
+  'global.colors.text': {
+    description: 'The text color used inside the Paragraph.',
+    type: 'string | { dark: string, light: string }',
+    defaultValue: "{ dark: '#f8f8f8', light: '#444444' }",
+  },
+  'global.text': {
```
Comment (ericsoderberghp) [inline src/js/components/Paragraph/doc.js]: I thought the text sizing was always handled by `paragraph.<size>`?

**Pull request merged**